packages/ui-components/src/components/Select/Select.component.tsx

```diff
@@ -0,0 +1,406 @@
+  open?: boolean
+  /** LEGACY: Handler to be executed when the Select value changes. Here for backwards compatibility with apps based on older versions of Select. Use onChange instead. */
+  // eslint-disable-next-line no-unused-vars
+  onValueChange?: (...args: unknown[]) => unknown
+  /** A placeholder to render when no value has been selected. Default is "Select…". */
+  placeholder?: string
+  /** Whether a selection is required. Will show a small required marker next to the label. If no label is used, no marker will be visible. */
+  required?: boolean
+  /** A note to render below the Select toggle in case the selected value has been positively validated. Will set the visible state of the Select toggle to `valid`. */
+  successtext?: React.ReactNode
+  /** Whether long texts in options will be truncated with "…" or not. Default is false. The Select toggle label will always be truncated. */
+  truncateOptions?: boolean
+  /** Whether the Select was positively validated. Will show a green checkmark icon inside the Select toggle. */
+  valid?: boolean
+  /** The currently (pre-)selected value of the Select. Will trigger controlled mode. When setting `multiple` on the Select pass an Array instead of a string. */
+  value?: unknown
+  /** The label of the passed value or defaultValue. If you want to use controlled mode or pass as defaultValue in uncontrolled mode and additionally use labels for
+   *  human-readable SelectOptions you need to also pass the matching label for the passed value/defaultValue so that the Select component can render itself properly. */
+  valueLabel?: string
+  /** The semantic variant of the Select toggle button.*/
+  variant?: "default" | "primary" | "primary-danger" | "subdued"
+  /** Whether the Select toggle should consume the available width of its parent container (default), or render its "natural" width depending on the content and the currently selected value or state. */
+  width?: "full" | "auto"
+  /** Pass a custom classname to the wrapping element. This can be useful if you must add styling to the outermost wrapping element of this component, e.g. for positioning. */
+  wrapperClassName?: string
+}
+
+/** 
+  A Select component that can be configured to allow selecting a single item or multiple items.
+  Pass a `defaultValue` to render as an uncontrolled component that tracks its open state etc internally.
+*/
+export const Select = ({
+  ariaLabel = "",
+  children = null,
+  className = "",
+  defaultValue,
+  disabled = false,
+  error = false,
+  errortext = "",
+  helptext = "",
+  id = "",
+  invalid = false,
+  label,
+  loading = false,
+  multiple = false,
+  name,
+  onChange,
+  onValueChange,
+  placeholder = "Select…",
+  required = false,
+  successtext = "",
+  truncateOptions = false,
+  valid = false,
+  value,
+  valueLabel,
+  variant = "default",
+  width = "full",
+  wrapperClassName = "",
+  ...props
+}: SelectProps) => {
+  const isNotEmptyString = (
```

Comment:
Let's add a return type `boolean`. 

Imagine `!isNotEmptyString`. 🤪 But we don't have these cases anywhere, so naming is fine and makes the code more readable as is. 😅